content/essentials/list-rendering.mdx

```diff
@@ -0,0 +1,213 @@
+
+To access `<For>` , you can import it directly from Solid:
+
+```
+import { For } from "solid-js";
+```
+
+The sole property that `<For>` has is `each` , through which you can specify the data collection to loop over:
+
+```jsx
+<For each={data()}>
+  {(item, index) => 
+    // rendering logic for each element
+  }
+</For>
+```
+
+Between the `<For>` tags, the component requires a callback function which will dictate how each item in the data collection should be rendered.
+This structure of rendering function resembles the callback used within JavaScript's [`map`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) method, providing a familiar pattern to follow.
+The function receives two arguments:
+* `item`: represents the current item in the data collection that is being rendered over.
+* `index`: the current item's index within the collection.
+
+Within the rendering function, you create and return JSX elements that will represent each item in the DOM.
+It is also possible to access properties of the current `item` to dynamically set attributes or content of the JSX elements.
+The `index` can be used for various purposes, such as controlling styling, setting unique keys, or any other case that requires the position of an item within the collection.
+Additionally, [conditional rendering](/essentials/conditional-rendering) can be used to show how or whether certain elements are rendered based on conditions related to the `item` or `index`.
+
+```jsx
+<For each={data()}>
+  {(item, index) => 
+    <li classList={{ even: index % 2 === 0, odd: index % 2 !== 0 }}>
+      {item.name}
+    </li>
+  }
+</For>
+```
+
+In this example, the [`classList`](/essentials/class-style-bindings#classList) directive is used to conditionally apply the `even` and `odd` classes to the `li` elements based on the current index.
+
+Furthermore, within the `<For>` component, other functions can be used to compute values, manage side effects, or handle events.
+There's also the ability to nest other looping components within the original `<For>` component, or to employ array methods to manage nested data structures or create more complex layouts:
+
+```jsx
+function App() {
+  const categories = [
+    {
+      name: 'Fruits',
+      items: [{ name: 'Apple', selected: false }, { name: 'Banana', selected: false }]
+    },
+    {
+      name: 'Vegetables',
+      items: [{ name: 'Carrot', selected: false }, { name: 'Potato', selected: false }]
+    }
+    // ... more categories
+  ];
+
+  const toggleSelection = () => {
+    // function logic to toggle item selection
+  };
+
+  return (
+    <ul>
+      <For each={categories}>
+        {(category, categoryIndex) => (
+          <li>
+            {category.name}
+            <ul>
+              <For each={category.items}>
+                {(item, itemIndex) => (
+                  <li
```

Comment:
You can use `categoryIndex` and `itemIndex` as the `li` keys. You could also have the inner `li` keys be a combination of the 2 `categoryIndex.toString() + itemIndex.toString()`. This is just a best practice, not really necessary if the list isn't reactively changing its length or moving around its values